File: Java Advanced/3. Sets and Maps/Sets and Maps/src/HandsOfCards.java
import java.util.*;
import java.util.stream.Collectors;
import java.util.stream.Stream;

public class HandsOfCards {
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        Map<String, LinkedHashSet<String>> score = new LinkedHashMap<>();

        String[] line = s.nextLine().split(": ");

        while (!"JOKER".equals(line[0])) {
            String name = line[0];
            String[] temp = line[1].split(", ");
            LinkedHashSet<String> cards = new LinkedHashSet<>();
            Collections.addAll(cards, temp);

            if (!score.containsKey(name)) {
                score.put(name, cards);
            } else {
                LinkedHashSet<String> oldSet = score.get(name);
                oldSet.addAll(cards);
                score.put(name, oldSet);
            }
            line = s.nextLine().split(": ");
        }

        for (Map.Entry<String, LinkedHashSet<String>> entry : score.entrySet()) {
            int sum = 0;
            int number;
            int S = 4;
            int H = 3;
            int D = 2;
            int C = 1;
            for (String card : entry.getValue()) {
                char[] num = card.toCharArray();
                if (num.length == 3) {
                    number = 10;
                    if (num[num.length - 1] == 'S') {
                        sum = sum + (number * S);
                    } else if (num[num.length - 1] == 'H') {
                        sum = sum + (number * H);
                    } else if (num[num.length - 1] == 'D') {
                        sum = sum + (number * D);
                    } else if (num[num.length - 1] == 'C') {
                        sum = sum + (number * C);
                    }
                } else if (num.length < 3) {
                    String temp = num[0] + "";
                    if (temp.equals("A")) {
                        number = 14;
                    } else if (temp.equals("K")) {
                        number = 13;
                    } else if (temp.equals("Q")) {
                        number = 12;
                    } else if (temp.equals("J")) {
                        number = 11;
                    } else {
                        number = Integer.parseInt(temp);
                    }
                    if (num[num.length - 1] == 'S') {
                        sum = sum + (number * S);
                    } else if (num[num.length - 1] == 'H') {
                        sum = sum + (number * H);
                    } else if (num[num.length - 1] == 'D') {
                        sum = sum + (number * D);
                    } else if (num[num.length - 1] == 'C') {
                        sum = sum + (number * C);
                    }
                }

            }
            System.out.println(entry.getKey() + ": " + sum);
        }

    }
}

File: soft uni prgramming fundamentals/7. Dictionaries, Lambda Expressions and LINQ/Dictionaries/HandsOfCards/HandsOfCards.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace HandsOfCards
{
    class HandsOfCards
    {
        static void Main(string[] args)
        {
            string[] input = Console.ReadLine().Split(":").ToArray();
            Dictionary<string, List<string>> result = new Dictionary<string, List<string>>();

            while (input[0] != "JOKER")
            {
                string playerName = input[0];
                List<string>cards = input[1].Split(new char[] { ',', ' ' },StringSplitOptions.RemoveEmptyEntries).ToList();
                cards = cards.Distinct().ToList();
                             

                if (playerName != ":")
                {
                    if (result.ContainsKey(playerName) == false)
                    {
                        result.Add(playerName, cards);
                    }
                    result[playerName].AddRange(cards);
                    result[playerName] = result[playerName].Distinct().ToList(); ;
                }
                      
                  

                input = Console.ReadLine().Split(":").ToArray();
            }

            foreach (var item in result)
            {

                int sum = 0;
                int number; ;
                int s = 4;
                int h = 3;
                int d = 2;
                int c = 1;
                foreach (var card in item.Value)
                {
                    for (int i = 0; i < 1; i++)
                    {
                        char[] num = card.ToCharArray();

                        if (num.Length == 3)
                        {
                            number = 10;
                            if (num[num.Length - 1] == 'S')
                            {
                                sum = sum + (number * s);
                            }
                            else if (num[num.Length - 1] == 'H')
                            {
                                sum = sum + (number * h);
                            }
                            else if (num[num.Length - 1] == 'D')
                            {
                                sum = sum + (number * d);
                            }
                            else if (num[num.Length - 1] == 'C')
                            {
                                sum = sum + (number * c);
                            }
                        }
                        else if (num.Length < 3)
                        {
                            string temp = num[0].ToString();
                            if (temp == "A")
                            {
                                number = 14;
                            }
                            else if (temp == "K")
                            {
                                number = 13;
                            }
                            else if (temp == "Q")
                            {
                                number = 12;
                            }
                            else if (temp == "J")
                            {
                                number = 11;
                            }
                            else
                            {
                                number = int.Parse(temp);
                            }
                            if (num[num.Length - 1] == 'S')
                            {
                                sum = sum + (number * s);
                            }
                            else if (num[num.Length - 1] == 'H')
                            {
                                sum = sum + (number * h);
                            }
                            else if (num[num.Length - 1] == 'D')
                            {
                                sum = sum + (number * d);
                            }
                            else if (num[num.Length - 1] == 'C')
                            {
                                sum = sum + (number * c);
                            }
                        }

                    }
                }
                Console.WriteLine($"{item.Key}: {sum}");
            }
        }
    }
}
